config/cobbler/settings.yaml

```diff
@@ -619,8 +618,21 @@ mongodb:
   host: "localhost"
   port: 27017
 
+# Set to true to cache of some internal operations can speed up their execution, but can slow down
+# editing objects.
+cache_enabled: false
+
 # This should contain the protocol over which the autointall-file is available.
 # Valid values are: http, https
 # This is setting does not setup your api for https it just changes the way the url for your profiles and systems are
 # generated.
 autoinstall_scheme: "http"
+
+# Set to true to speed up the start of the Cobbler. When storing collections as files, the directory with the names
+# of the collection elements will be scanned without reading and parsing the files themselves. In the case of storing
+# collections in the database, a projection query is made that includes only the names of the collection elements.
+# The first time an attribute of an element other than a name is accessed, a full read of all other attributes will be
+# performed, and a recursive full read of all elements on which this element depends. At startup, a background task is
+# also launched, which, when idle, fills in all the properties of the elements of the collections.
+# Suitable for configurations with a large number of elements placed on a slow device (HDD, network).
+lazy_start: false
```

Comment:
My problem with this method of loading is atm that my plan was at some point to transition the file names to the UUIDs, so we gain the benefit of not having to rename the file on an "item_rename" operation. However, your change (as brilliant as the idea is) would prevent that from happening. Do you see it realistic in a follow-up PR to introduce my change and still keep your `lazy_start`?

Reply:
I think this can be done if the key for `collection.listing` is not the name, but uid. Then the search by name will be somewhat slower, so it will need to be optimized. Optimization can be done, for example, by creating additional indexes on non-key fields, or by caching search results.

Reply:
Okay perfect. Then let's skip this problem for now. This is anyway something for 3.5.x and not for 3.4.x

Reply:
I thought again, and it seems to me that there will be no slowdown, but most likely there will be an acceleration. Much of the search is driven by finding a related object by name. But if instead of a name, for example, in `system.profile` or `profile.parent` there is uid, then most of the search will be by uid.
Now it seems to me that the idea of file names and association of objects by uid will greatly simplify and speed up everything. In this case, the code associated with `lazy_start` will not change.